external-builds/pytorch/patches/pytorch/v2.7.0/pytorch/hipified/0012-Enable-USE_ROCM-and-fix-assorted-issues.patch

```diff
@@ -42,25 +39,8 @@ index f3fee2f7ff..6086450e3e 100644
  cmake_dependent_option(USE_STATIC_NCCL "Use static NCCL" OFF "USE_NCCL" OFF)
  cmake_dependent_option(USE_SYSTEM_NCCL "Use system-wide NCCL" OFF "USE_NCCL"
                         OFF)
-diff --git a/aten/src/ATen/CMakeLists.txt b/aten/src/ATen/CMakeLists.txt
-index 74e464dda4..009859de63 100644
---- a/aten/src/ATen/CMakeLists.txt
-+++ b/aten/src/ATen/CMakeLists.txt
-@@ -382,12 +382,6 @@ endif()
-     ${native_quantized_hip_hip}
-     ${native_transformers_hip_hip} ${native_transformers_src_hip_hip}
-   )
--  if(WIN32) # Windows doesn't support Composable Kernels
--    file(GLOB native_hip_bgemm "native/hip/bgemm_kernels/*.hip")
--    file(GLOB native_hip_ck "native/hip/ck*.hip")
--    exclude(ATen_HIP_SRCS "${ATen_HIP_SRCS}"
--      ${native_hip_bgemm} ${native_hip_ck})
--  endif()
-   # TODO: Codegen separate files for HIP and use those (s/cuda_generated_sources/hip_generated_sources)
-   list(APPEND all_hip_cpp
-     ${native_nested_hip_cpp}
```

Comment:
This diff is _removed_ from the patch, as is the "* Composable Kernels generate linker errors if sources are excluded" text from the commit description.

Reply:
(net result is that CMake code removing files _stays_ in PyTorch... so many double negatives)

Reply:
Interesting. I was facing errors without this diff. Do you still face the errors when trying to build with aotriton enabled?

Reply:
I haven't tried with aotriton, but I did verify that the changes from https://github.com/pytorch/pytorch/commit/3e78c9e967f154301937d4de1f5d9b4ab2cb5d84 do now get reflected in `aten\src\ATen\hip\HIPBlas.cpp` and the linker errors go away.

Reply:
Ahh I see> Yeah i missed cherry-picking  that commit in.. makes sense now!